docker-compose.yml

```diff
@@ -0,0 +1,111 @@
+version: '3.5'
+services:
+  app: &app
+    build:
+      context: .
+      target: hyrax-engine-dev
+      args:
+        BUNDLE_WITHOUT: test
+    image: hyrax-engine-dev
+    stdin_open: true
+    tty: true
+    environment:
+      - DATABASE_URL=postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5
+      - FCREPO_PORT=8080
+      - FCREPO_HOST=fcrepo
+      - FCREPO_REST_PATH=fcrepo/rest
+      - HYRAX_ENGINE_PATH=/app-data/samvera/hyrax-engine
+      - RACK_ENV=development
+      - RAILS_ENV=development
+      - REDIS_HOST=redis
+      - SOLR_URL=http://solr:8983/solr/hyrax
+    depends_on:
+      - db_migrate
+      - fcrepo
+      - memcached
+      - postgres
+      - redis
+      - solr
+    ports:
+      - 3000:3000
+    volumes:
+      - .dassie:/app-data/samvera/hyrax-webapp
+      - .:/app-data/samvera/hyrax-engine
+
+  db_migrate:
+    image: hyrax-engine-dev
+    environment:
+      - DATABASE_URL=postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5
+      - FCREPO_PORT=8080
+      - FCREPO_HOST=fcrepo
+      - FCREPO_REST_PATH=fcrepo/rest
+      - HYRAX_ENGINE_PATH=/app-data/samvera/hyrax-engine
+      - SOLR_URL=http://solr:8983/solr/hyrax
+    command:
+      - /app-data/samvera/db-wait.sh
+      - postgres:5432
+      - /app-data/samvera/db-wait.sh
+      - fcrepo:8080
+      - /app-data/samvera/db-wait.sh
+      - solr:8983
+      - bundle
+      - exec
+      - rails
+      - db:setup
+      - db:migrate
+      - db:seed
+    depends_on:
+      - postgres
+    volumes:
+      - .dassie:/app-data/samvera/hyrax-webapp
+      - .:/app-data/samvera/hyrax-engine
+
+  postgres:
+    image: postgres:latest
+    restart: always
+    environment:
+      - POSTGRES_USER=hyrax_user
+      - POSTGRES_PASSWORD=hyrax_password
+      - POSTGRES_DB=hyrax
+      - POSTGRES_HOST_AUTH_METHOD=trust
+    ports:
+      - "5432:5432"
+    volumes:
+      - db:/var/lib/postgresql/data
+
+  fcrepo:
+    image: cbeer/fcrepo4:4.7
+    volumes:
+      - fcrepo:/data
+    ports:
+      - 8080:8080
+    environment:
+      - JAVA_OPTS=${JAVA_OPTS} -Dfcrepo.modeshape.configuration="classpath:/config/file-simple/repository.json" -Dfcrepo.object.directory="/data/objects" -Dfcrepo.binary.directory="/data/binaries"
+
+  memcached:
+    image: bitnami/memcached
+    ports:
+      - '11211:11211'
+
+  redis:
+    image: redis:5-alpine
+    volumes:
+      - redis:/data
+
+  solr: &solr
+    image: solr:7
+    ports:
+      - 8983:8983
+    command:
+      - solr-precreate
+      - hyrax
+      - /opt/solr/server/configsets/hyraxconf
```

Comment:
I know you're not doing test env yet, but I ran across this the other night and haven't had a chance to try it out yet. It should create 2 cores.

```yaml
    - bash
    - "-c"
    - "precreate-core hyrax /opt/solr/server/configsets/hyraxconf; precreate-core hyrax-test /opt/solr/server/configsets/hyraxconf; exec solr -f"
```

Reply:
putting this in my pocket for later.